x-pack/plugins/rule_registry/server/rule_data_plugin_service/rule_data_plugin_service.test.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { loggerMock } from '@kbn/logging/mocks';
+import { RuleDataService } from './rule_data_plugin_service';
+import { elasticsearchServiceMock } from 'src/core/server/mocks';
+import { AlertConsumers } from '@kbn/rule-data-utils/alerts_as_data_rbac';
+
+import { Dataset } from './index_options';
+jest.mock('../rule_data_client/rule_data_client');
```

Comment:
@weltenwort Before I open this PR for review, I would like your input regarding what's the recommended way to mock a class constructor. Later in my test I do 
```
expect(
        jest.requireMock('../rule_data_client/rule_data_client').RuleDataClient
      ).toHaveBeenCalled();
```
and it works. I am just wondering if there's a better way.

Reply:
Looks like there already is a mock implementation of the rule data client in https://github.com/elastic/kibana/blob/bfa8b3c9b6f3e7d58335cbfe991de519e2f087ec/x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.mock.ts#L21-L43. So maybe you could use that like this:

```typescript
jest.mock('../rule_data_client/rule_data_client', () => ({
  RuleDataClient: jest.fn().mockImplementation(createRuleDataClientMock),
}));
```

And further down I saw you're calling `jest.mock` again. Instead I would recommend to just import the `RuleDataClient` normally. The jest mocking mechanism will make sure it's replaced with the mock.

Similarly, it looks like you're calling `jest.mock('./resource_installer', () => {` inside of a test case. I think it's safer to only call `jest.mock` on the top level of the file as it needs to be evaluated before any import.